### PR TITLE
Implement rethrow

### LINKED ISSUE #60: Implement rethrow
Something like

```scala
final def rethrow[E1 >: E, B](implicit ev: A <:< Either[E1, B]): BIO[E1, B]
```

to go from `BIO[E, Either[E, B]]` to `BIO[E, B]`

Prior art: 
- https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/MonadError.scala#L39 (rethrow)
- https://github.com/zio/zio/blob/master/core/shared/src/main/scala/zio/ZIO.scala#L77 (absolve)

### PR DESCRIPTION
**Pull request by bszwej**
Fixes #60

### CODE REVIEW

**Comment by Avasil** [comment]
Looks great, thank you @bszwej 
I feel like it could use a method on companion objects (BIO, Task (specialized to Throwable), UIO (specialized to Nothing) as well - would you mind adding it there?

File: core/shared/src/main/scala/monix/bio/Task.scala [hunk 1]
```diff
@@ -236,6 +236,12 @@ object Task {
   def racePair[A, B](fa: Task[A], fb: Task[B]): Task[Either[(A, Fiber[Throwable, B]), (Fiber[Throwable, A], B)]] =
     BIO.racePair(fa, fb)
 
+  /**
+    * @see See [[monix.bio.BIO.rethrow]]
+    */
+  final def rethrow[A, B](fa: Task[A])(implicit ev: A <:< Either[Throwable, B]): Task[B] =
```
Comment (Avasil) [inline core/shared/src/main/scala/monix/bio/Task.scala]: The signature could be
```scala
def rethrow[A](fa: Task[Either[Throwable, A]]): Task[A]
```

Similarly with `UIO`

And I was thinking about including it in `BIO` companion object as well so we can call both `fa.rethrow` and `BIO.rethrow(fa)`

Comment (bszwej) [reply core/shared/src/main/scala/monix/bio/Task.scala]: Thanks for the suggestion. 
Done :)!

**Comment by Avasil** [comment]
Thank you for the contribution @bszwej

**Pull request merged**